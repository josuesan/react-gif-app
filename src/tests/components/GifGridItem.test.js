import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGridItem from '../../components/GifGridItem';

describe('Test to grid gifs', () => {

  const title = 'Un titulo';
  const url = 'https://localhost.com/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should be renderer', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a parragraph with a text', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should have a img with source and alt text', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have animeted css', () => {
    const div = wrapper.find('div');
    expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
  })

});
