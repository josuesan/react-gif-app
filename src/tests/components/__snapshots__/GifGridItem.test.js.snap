// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to grid gifs should be renderer 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost.com/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
